### DISCUSSION TITLE
请问如何在doris中实现字符串中文分词功能？

### DISCUSSION THREAD

**Discussion by JamesLane72** [apache/doris#30786] [A - General / Q&A]
doris-1.2.7。doris本身似乎并没有中文分词函数，我尝试添加java udf的方式来进行中文分词，但是发现同一个udf函数在hive中能添加以及正常运行，在doris中却无法添加，会抛出异常，内容如下：
    Method 'evaluate' in class 'com.xxx.udf.JiebaAnalyzerSeg' does not support non-scalar type 'ARRAY<TEXT>'.
从报错信息来看，经过多次测试，发现似乎是由于Jieba分词中使用了Text类型的对象而导致的。
请问还有其它方法在doris中实现字符串中文分词功能吗？

**Comment by LemonLiTree** [comment]
升级到2.0.4 有分词函数 也有对应的倒排索引

**Reply by JamesLane72** [reply]
好的，感谢解答！

**Comment by JamesLane72** [comment]
经过反复测试，发现是evaluate方法返回list类型而导致的报错，改为返回string类型即可。
之前报错信息为：
Method 'evaluate' in class 'com.xxx.udf.JiebaAnalyzerSeg' does not support non-scalar type 'ARRAY\<TEXT\>'.
文档中有说明ArrayList<Type>与array<Type>的类型对应关系，以为是支持返回list类型而不支持text，但是从测试结果看来doris-1.2.7 java UDF不支持返回List<Type>，不知是否正确。

**Reply by LemonLiTree** [reply]
right
